relay-server/src/actors/store.rs

```diff
@@ -199,51 +199,101 @@ impl StoreService {
-                event_type = &event_item.ty().to_string()
-            );
-        } else if !attachments.is_empty() {
-            relay_log::trace!("Sending individual attachments of envelope to kafka");
-            for attachment in attachments {
-                let attachment_message = KafkaMessage::Attachment(AttachmentKafkaMessage {
-                    event_id: event_id.ok_or(StoreError::NoEventId)?,
-                    project_id: scoping.project_id,
-                    attachment,
-                });
+        let remote_addr = envelope.meta().client_addr().map(|addr| addr.to_string());
 
-                self.produce(topic, scoping.organization_id, attachment_message)?;
-                metric!(
-                    counter(RelayCounters::ProcessingMessageProduced) += 1,
-                    event_type = "attachment"
-                );
+        let kafka_messages = Self::extract_kafka_messages(
+            event_item,
+            event_id.ok_or(StoreError::NoEventId)?,
+            scoping,
+            start_time,
+            remote_addr,
+            attachments,
+        );
+
+        for message in kafka_messages {
+            if let Ok(()) = self.produce(topic, scoping.organization_id, &message) {
+                // If message was succesfully sent, make the right log message.
+                match message {
+                    KafkaMessage::Attachment(_) => {
+                        metric!(
+                            counter(RelayCounters::ProcessingMessageProduced) += 1,
+                            event_type = "attachment"
+                        );
+                    }
+                    _ => {
+                        if let Some(event_item) = event_item {
+                            metric!(
+                                counter(RelayCounters::ProcessingMessageProduced) += 1,
+                                event_type = &event_item.ty().to_string()
+                            );
+                        }
+                    }
+                }
             }
         }
 
         Ok(())
     }
 
+    fn extract_kafka_messages(
+        event_item: Option<&Item>,
+        event_id: EventId,
+        scoping: Scoping,
+        start_time: Instant,
+        remote_addr: Option<String>,
+        attachments: Vec<ChunkedAttachment>,
+    ) -> impl Iterator<Item = KafkaMessage> {
+        // There might be a better way to do this:
+        let (individual_attachments, inline_attachments) = match event_item {
+            Some(event_item) => {
+                if matches!(event_item.ty(), ItemType::Transaction) {
+                    (attachments, vec![])
+                } else {
+                    (vec![], attachments)
+                }
+            }
+            None => (attachments, vec![]),
+        };
+
+        let project_id = scoping.project_id;
+
+        let event_iterator = event_item
+            .map(|event_item| {
+                KafkaMessage::Event(EventKafkaMessage {
```

Comment:
Let's restore these log lines:

```suggestion
                relay_log::trace!("Sending event item of envelope to kafka");
                KafkaMessage::Event(EventKafkaMessage {
```

Reply:
Right now the event and the individual attachments are being produced in the same for-loop so I can't recreate the log lines as they were, I could put some logs where we record the metrics.

Although the log messages were dealing with the logic in match statement at the start of the `fn extract_kafka_messages_for_event`, then perhaps we should write a separate log for each of the three branches there instead